pkg/vf/virtionet_test.go

```diff
@@ -0,0 +1,100 @@
+package vf
+
+import (
+	"fmt"
+	"math/rand"
+	"net"
+	"os"
+	"path/filepath"
+	"strings"
+	"testing"
+
+	"github.com/crc-org/vfkit/pkg/config"
+)
+
+func testConnectUnixgram(t *testing.T) error {
+	unixSocketPath := filepath.Join("/tmp", fmt.Sprintf("vnet-test-%x.sock", rand.Int31n(0xffff))) //#nosec G404 -- no need for crypto/rand here
+	addr, err := net.ResolveUnixAddr("unixgram", unixSocketPath)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+	l, err := net.ListenUnixgram("unixgram", addr)
+	if err != nil {
+		t.Errorf("Unexpected error: %v", err)
+	}
+
+	defer l.Close()
+	defer os.Remove(unixSocketPath)
+
+	dev := &VirtioNet{
+		&config.VirtioNet{
+			UnixSocketPath: unixSocketPath,
+		},
+		&net.UnixAddr{},
+	}
+
+	return dev.connectUnixPath()
+}
+
+func TestConnectUnixPath(t *testing.T) {
+	t.Run("Successful connection - no error", func(t *testing.T) {
+		err := testConnectUnixgram(t)
+		if err != nil {
+			t.Fatalf("Unexpected error: %v", err)
+		}
+	})
+
+	t.Run("Failed connection - End socket longer than 104 bytes", func(t *testing.T) {
+		// Retrieve HOME env variable (used by the os.UserHomeDir)
```

Comment:
Ah right, kind of messy to do the 100 bytes test with current `main` :/ This will need to be modified once https://github.com/crc-org/vfkit/pull/195 gets in.